ruby/ql/lib/codeql/ruby/ApiGraphs.qll

```diff
@@ -266,6 +266,29 @@ module API {
   /** A node corresponding to the method being invoked at a method call. */
   class MethodAccessNode extends Node, Impl::MkMethodAccessNode {
     override string toString() { result = "MethodAccessNode " + tryGetPath(this) }
+
+    /** Gets the call node corresponding to this method access. */
+    DataFlow::CallNode getCallNode() { this = Impl::MkMethodAccessNode(result) }
+  }
+
+  /**
+   * An API entry point.
+   */
+  abstract class EntryPoint extends string {
+    bindingset[this]
+    EntryPoint() { any() }
+
+    /** Gets a data-flow node corresponding to a use-node for this entry point. */
+    DataFlow::LocalSourceNode getAUse() { none() }
+
+    /** Gets a data-flow node corresponding to a def-node for this entry point. */
+    DataFlow::Node getARhs() { none() }
```

Comment:
It's not really clear what right-hand-side means; the right-hand of what? The name suggest there should also be a left-hand-side. Wouldn't `getADef` be a better name, since there is already a `getAUse` method.

Reply:
I had a similar question [here](https://github.com/github/codeql/pull/7819#discussion_r801210664) about `API::Node.getARhs()`. I still believe it's a confusing name in the context of Ruby.

Reply:
`getASource`?

Reply:
It's a metaphor, relating to a concept that's pretty well-established in with compilers/program analysis. The right-hand side of an assignment is the value "flowing into" something.

I think you'd come to regret the names you're suggesting if you actually tried to used the predicate, rather than just looking at its definition. Please keep in mind that you can't always evaluate an API just by looking at its definition; you have to consider what it looks like at the use site.

`getADefNode`:
1. Same problem as `getADefinition`, a _definition_ of a parameter is an argument?!
2. API graphs have a concept called "def-nodes", and the predicate doesn't actually return a def-node, it returns a data-flow node (when the receiver is a def-node).

`getASource`:
1. It's not clear if it involves transitive flow. Does it correspond to `getAValueReachingRhs` or just `getARhs`? With `getARhs` it's pretty obvious that transitive flow is not implied. `getAnImmediateSource` might be OK though.
2. It doesn't return `LocalSourceNode`, though the name suggests that it might.
3. You end up using `getASource` when defining a sink, and you'll get confused review comments suggesting that you made a mistake there. For example:
```codeql
override predicate isSink(DataFlow::Node node) {
  node = API::moduleImport("foo").getMember("bar").getParameter(0).getASource() // wait, source or sink?
}
```

----

With that being said, I recognise there's a lot of backlash against the current name, and I'm open to finding a better name. But the new name has to actually be better.

Also, it's a little awkward having the discussion in this PR since API graphs have been around for a long time already, it's just that Python and Ruby didn't have the full version. It would be nice if you'd also recognize that you are suggesting a rename of an existing predicate with lots of uses. It's not some new thing I invented on the spot here.